src/crypto/sha256.cpp

```diff
@@ -716,6 +716,63 @@ void CSHA256::Midstate(unsigned char hash[OUTPUT_SIZE], uint64_t* len, unsigned
     }
 }
 
+std::vector<unsigned char> CSHA256::Save() {
+   std::vector<unsigned char> result;
+   result.reserve(40 + (bytes % 64));
+
+   for(auto i = 0; i < 8; ++i) {
+     result.push_back(static_cast<unsigned char>(s[i] >> 24));
```

Comment:
You can use `WriteBE32` here instead.

Reply:
I would have to do an allocation rather than a reservation, but I think that would be a good idea.  That would also let me use `memcpy` below.